sql/catalyst/src/main/scala/org/apache/spark/sql/types/CharType.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.types
+
+import scala.math.Ordering
+import scala.reflect.runtime.universe.typeTag
+
+import org.apache.spark.annotation.Experimental
+import org.apache.spark.unsafe.types.UTF8String
+
+@Experimental
+case class CharType(length: Int) extends AtomicType {
```

Comment:
Shall we add `SPARK-6412 (Add Char support in dataTypes)` into the PR title as a secondary JIRA?
- SPARK-6412 was resolved as `Later`.

Reply:
The PR description says `To be safe, this PR doesn't add char/varchar type to the core type system.`, but this adds `CharType` into `org.apache.spark.sql.types`. Do you revise a little the PR description? It seems that you wanted to mention the SQL execution parts.

Reply:
It's already there: https://github.com/apache/spark/pull/30412/files#diff-aacc5ed42589c636615a3c09a44fa6a5248195242c9fbe0e996db17471cd35fdL70

I just moved it to a new file and remove the parent class.

Reply:
And type system means a lot more, e.g. expression type check, InternalRow.get, etc. Maybe I should say "type system framework"?

Reply:
PR description updated